doc: doc release PR for 0.8.0 (#312)
nrfconnect/sdk-nrf-bm: doc/nrf-bm/release_notes/release_notes_0.8.0.rst

.. _nrf_bm_release_notes_080:

Release Notes for |BMlong| v0.8.0
#################################

.. contents::
   :local:
   :depth: 2

> |BMshort| v0.8.0 is the public launch release of this solution, demonstrating a set of samples for the `nRF54L15 DK`_ with build targets supporting the following SoCs:

Reviewer: ```suggestion
|BMshort| v0.8.0 is the public launch release of this product, demonstrating a set of samples for the `nRF54L15 DK`_ with build targets supporting the following SoCs:
```
At the bootom of this file, we refer to this being a product: `First public release of product documentation.`

* `nRF54L15`_
* `nRF54L10`_

> * `nRF54L05`_

Reviewer 2: Should it say "emulated" or similar?

The release includes a prototype version of the S115 SoftDevice - a proprietary Bluetooth® LE stack to be used on the nRF54L15 DK.

IDE and tool support
********************

`nRF Connect extension for Visual Studio Code <nRF Connect for Visual Studio Code_>`_ is the recommended IDE for |BMshort| v0.8.0.
See the :ref:`install_nrf_bm` section for more information about supported operating systems and toolchain.

Supported boards
****************

* PCA10156 (`nRF54L15 DK`_)

Changelog
*********

The following sections provide detailed lists of changes by component.

SDK installation
================

* The default way for installing |BMshort| is through the |nRFVSC| user interface, using either the :guilabel:`Pre-packaged SDK & Toolchain` option or a GitHub installation.
  For detailed steps, see :ref:`cloning_the_repositories_nrf_bm`.

S115 SoftDevice
===============

* Updated the S115 SoftDevice version to ``s115_9.0.0-3.prototype``.
  The SoftDevice comes in three variants to support different SoCs of the nRF54L family: nRF54L15, nRF54L10, and nRF54L05.

  * :file:`s115_9.0.0-3.prototype_nrf54l15_softdevice`
  * :file:`s115_9.0.0-3.prototype_nrf54l10_softdevice`

>   * :file:`s115_9.0.0-3.prototype_nrf54l05_softdevice`

Author: I meant file names. Added `.`hex``.

  For more details about this release of the SoftDevice, see :ref:`s115_docs`.

> Softdevice Handler

Reviewer 2: ```suggestion
SoftDevice Handler
```

==================

* Added support for seeding the SoftDevice RNG when requested.
* Fixed the maximum MTU Kconfig option.

Boards
======

* Renamed the ``bm_nrf54l15dk`` board variants:

  * ``bm_nrf54l15dk/nrf54l05/softdevice_s115`` to ``bm_nrf54l15dk/nrf54l05/s115_softdevice``
  * ``bm_nrf54l15dk/nrf54l10/softdevice_s115`` to ``bm_nrf54l15dk/nrf54l10/s115_softdevice``
  * ``bm_nrf54l15dk/nrf54l15/softdevice_s115`` to ``bm_nrf54l15dk/nrf54l15/s115_softdevice``

* Added the following ``bm_nrf54l15dk`` board variants for MCUboot (DFU-enabled board variants):

  * ``bm_nrf54l15dk/nrf54l05/softdevice_s115`` to ``bm_nrf54l15dk/nrf54l05/s115_softdevice/mcuboot``
  * ``bm_nrf54l15dk/nrf54l10/softdevice_s115`` to ``bm_nrf54l15dk/nrf54l10/s115_softdevice/mcuboot``
  * ``bm_nrf54l15dk/nrf54l15/softdevice_s115`` to ``bm_nrf54l15dk/nrf54l15/s115_softdevice/mcuboot``

> * Removed the template board variants.

Reviewer 3: Also removed the no_softdevice variants of the bm_nRF54l15dk

DFU
===

* The DFU subsystem is now introduced in the SDK.
  For more details, see :ref:`bm_dfu`.

Logging
=======

* Fixed an issue with the minimal log mode.

Drivers
=======

* Added the following drivers:

  * The :ref:`driver_lpuarte` driver.

>   * The SoftDevice-based flash driver.

Reviewer 3: Perhaps something like "this is exclusively added for DFU and is not expected to be used by other parts of the application. Instead have a look at ZMS and the storage library".

>   * The SoftDevice-based flash driver.

Author: So this is now changed to 

  * A SoftDevice-based flash driver implementing the Zephyr flash interface.
    This is added exclusively for DFU and is not expected to be used by other parts of the application.
    Instead, use the :ref:`lib_bm_zms` and ``bm_storage`` libraries.

Libraries
=========

* Added the following libraries:

  * The ``bm_zms`` storage. TODO link when ready

>   * The ``Storage`` library.

Author: @MirkoCovizzi  ?

>   * A Zephyr queue functionality.

Reviewer 3: Yes, I think having a note about that would be good.

>   * A Zephyr queue functionality.

Reviewer 4: Yes, but drop Firmware Loader, it's a dependency for MCUmgr. I did a quick check now.

..
* ``ble_adv`` library:

  * Changed the default advertising name to ``nRF_BM_device``.
  * Corrected the :kconfig:option:`BLE_ADV_DIRECTED_ADVERTISING_HIGH_DUTY` Kconfig option.
..
* ``ble_conn_params library`` library:

>   * Added support for asymmetric data length.

Reviewer 4: Yes. DLE. Could it prove problematic you think?

>   * Added support for asymmetric data length.

Author: What is the final decision about this point then? @anhmolt  @seko-nordic  @bjorn-tore-taraldsen

>   * Added support for asymmetric data length.

Reviewer: I suggest we don't mention it here. Even if it is technically supported, it brings no benefit.

>   * Added support for asymmetric data length.

Reviewer 4: Yes, let's remove it.

  * Renamed slave latency to peripheral latency according to specification.

  * Fixed:

    * Data length timeout issue and the :c:func:`data_length_get` function returning wrong value.
    * Incorrect stated maximum ATT MTU value.
    * Removed unsupported bitmask from PHY update.
..
* ``bm_buttons`` library:

  * Added the use of nrfx critical section API.

Samples
=======

* General changes:

  * All samples now use the logging module instead of ``printk``.
  * Updated the main loop sleep instructions as the :c:func:`sd_app_evt_wait` function is deprecated.

>   * Fixed processing of deferred logging before returning from main.

Reviewer: Looking at the samples, they have all been changed to not return from main.

>   * Fixed processing of deferred logging before returning from main.

Author: Added as in the comment above. Just replaced 'sitting' with 'stored'. I will not resolve this thread yet in case someone has a different idea to handle this.

>   * Fixed processing of deferred logging before returning from main.

Author: done

Bluetooth samples
-----------------

* Added the :ref:`ble_pwr_profiling_sample` sample.
..
* :ref:`ble_cgms_sample` sample:

  * Fixed an issue where a PHY update request was responded twice.
..
* :ref:`ble_hrs_sample` sample:

>   * Removed the unused Coded PHY overlay.

Reviewer 3: ```suggestion
  * Removed the unused Coded PHY Kconfig fragment.
```

..
* :ref:`ble_nus_sample` sample:

  * Added support for Low Power UARTE (LPUARTE).
  * Fixed the pin configuration to use the pins from the ``board_config.h`` board header.

Peripheral samples
------------------

* Added the following peripheral samples:

  * :ref:`bm_zms_sample` sample.
  * :ref:`bm_lpuarte_sample` sample.
  * :ref:`storage_sample` sample.

MCUboot samples
---------------

* Added the following samples:

  * :ref:`ble_mcuboot_recovery_entry_sample` sample.
  * :ref:`ble_mcuboot_recovery_retention_sample` sample.

Known issues and limitations
============================

* Some issues are observed when using iPhone as the peer during testing of the Bluetooth samples.
* Some issues are observed when using Linux with the Bluetooth Low Energy app in nRF Connect for Desktop as the peer during testing of the Bluetooth samples.

> * Bluetooth LE pairing and bonding is not supported.

Author: done

.. _nrf_bm_release_notes_080:

Release Notes for |BMlong| v0.8.0
#################################

.. contents::
   :local:
   :depth: 2

|BMshort| v0.8.0 is the public launch release of this product, demonstrating a set of samples for the `nRF54L15 DK`_ with build targets supporting the following SoCs:

* `nRF54L15`_
* `nRF54L10`_ (emulated on the nRF54L15 DK)
* `nRF54L05`_ (emulated on the nRF54L15 DK)

The supported DK version is `0.9.3`_.

The release includes a prototype version of the S115 SoftDevice - a proprietary Bluetooth® LE stack to be used on the nRF54L15 DK.

Product maturity
****************

As this is version 0.8.0 of |BMshort|, please note that the solution is still under development.
While most modules and libraries are considered stable and are unlikely to undergo significant changes, adjustments might still be necessary.

> This includes potential modifications to the organization of the folder structure.

Reviewer 3: Add something about documentation, that we do not have documentation for all modules per now, though it will be added later.

Author: "The documentation for |BMshort| is still in development and might not cover all libraries and modules available in the codebase.
Some sections might be incomplete or lacking detailed information as content is being developed."

What do you think ? @eivindj-nordic

  * ``bm_nrf54l15dk/nrf54l10/softdevice_s115`` to ``bm_nrf54l15dk/nrf54l10/s115_softdevice/mcuboot``
  * ``bm_nrf54l15dk/nrf54l15/softdevice_s115`` to ``bm_nrf54l15dk/nrf54l15/s115_softdevice/mcuboot``

..

* Renamed the ``bm_nrf54l15dk`` board variants:

  * ``bm_nrf54l15dk/nrf54l05/softdevice_s115`` to ``bm_nrf54l15dk/nrf54l05/s115_softdevice``
  * ``bm_nrf54l15dk/nrf54l10/softdevice_s115`` to ``bm_nrf54l15dk/nrf54l10/s115_softdevice``
  * ``bm_nrf54l15dk/nrf54l15/softdevice_s115`` to ``bm_nrf54l15dk/nrf54l15/s115_softdevice``

..

* Fixed:

  * Fixed wrong RRAM size on the nRF54L10 emulation target from 1022 to 1012 KiB.

..

* Removed:

  * The template board variants.
  * The ``no_softdevice`` variants of the ``bm_nRF54l15dk``.

DFU
===

* The DFU subsystem is now introduced in the SDK.
  For more details, see :ref:`bm_dfu`.

Logging
=======

* Fixed an issue with the minimal log mode.

Drivers
=======

* Added the following drivers:

  * The :ref:`driver_lpuarte` driver.
  * A SoftDevice-based flash driver implementing the Zephyr flash interface.

Libraries
=========

* Added the following libraries:

>   * The ``bm_zms`` storage. TODO link when ready

Reviewer 5: will this TODO remain here?

Author: done

    * An issue where the ATT MTU could be set to a lower value than the current value.
    * An issue where the list of PHYs supported by the SoftDevice was not respected by the library when doing PHY updates.
    * Data length timeout issue and the :c:func:`data_length_get` function returning wrong value.
    * Incorrectly stated maximum ATT MTU value.

..

* ``bm_buttons`` library:

  * Added the use of nrfx critical section API.

Samples
=======

* General changes:

  * All samples now use the logging module instead of ``printk``.

  * Updated:
  
    * The main loop sleep instructions as the :c:func:`sd_app_evt_wait` function is deprecated.
    * Updated the main function flow to always end up in a loop with sleep instructions and deferred log processing instead of returning from the main function.
      This ensures that log messages are printed and that the device goes into sleep mode.
      Returning from main leaves the device in a state that prevents low power sleep and deferred log messages will be stored indefinitely in an internal buffer.

Bluetooth samples
-----------------

* Added the :ref:`ble_pwr_profiling_sample` sample.

..

* :ref:`ble_cgms_sample` sample:

>   * Fixed an issue where a PHY update request was responded to twice.

Author: ok

..

* :ref:`ble_hrs_sample` sample:

  * Removed the unused Coded PHY Kconfig fragment.

..

* :ref:`ble_nus_sample` sample:

  * Added support for Low Power UARTE (LPUARTE).

>   * Fixed the pin configuration to use the pins from the ``board_config.h`` board header.

Reviewer 3: After #333: 
* Fixed an issue in the :c:func:`ble_nus_data_send` function where passing NULL as the nus instance could cause a segmentation fault.

Peripheral samples
------------------

* Added the following peripheral samples:

  * :ref:`bm_zms_sample` sample.
  * :ref:`bm_lpuarte_sample` sample.
  * :ref:`storage_sample` sample.

MCUboot samples
---------------

* Added the following samples:

  * :ref:`ble_mcuboot_recovery_entry_sample` sample.
  * :ref:`ble_mcuboot_recovery_retention_sample` sample.

Known issues and limitations
============================

* Some issues are observed when using iPhone as the peer during testing of the Bluetooth samples.
* Some issues are observed when using Linux with the Bluetooth Low Energy app in nRF Connect for Desktop as the peer during testing of the Bluetooth samples.
* Bluetooth LE pairing and bonding is not supported.

> * The samples are not optimized for power consumption.

Reviewer 6: Adjust to:
"The samples are not optimized for power consumption unless explicitly stated"

      This ensures that log messages are printed and that the device goes into sleep mode.
      Returning from main leaves the device in a state that prevents low power sleep and unhandled deferred log messages are not processed.

Bluetooth samples
-----------------

* Added the :ref:`ble_pwr_profiling_sample` sample.

..

* :ref:`ble_cgms_sample` sample:

  * Fixed an issue with responding twice to a PHY update request.

..

* :ref:`ble_hrs_sample` sample:

  * Removed the unused Coded PHY Kconfig fragment.

..

* :ref:`ble_nus_sample` sample:

  * Added support for :ref:`driver_lpuarte`.

..

  * Fixed:

    * The pin configuration to use the pins from the :file:``board_config.h`` board header.
    * An issue in the :c:func:`ble_nus_data_send` function where passing NULL as the NUS instance could cause a segmentation fault.

Peripheral samples
------------------

* Added the following samples:

  * :ref:`bm_zms_sample` sample.
  * :ref:`bm_lpuarte_sample` sample.
  * :ref:`storage_sample` sample.

DFU samples
-----------

* Added the following samples:

>   * :ref:`ble_mcuboot_recovery_entry_sample` sample that demonstrates how to activate DFU mode on the device using a mobile application and execute DFU over Bluetooth LE.

Reviewer 6: ```suggestion
  * :ref:`ble_mcuboot_recovery_entry_sample` sample that demonstrates how to activate DFU mode on the device using Bluetooth Le on a mobile application.
```

Reviewer 6: Det focus here should be how to enter DFU mode, not that you could execute DFU (over whatever transport)

  * :file:`s115_9.0.0-3.prototype_nrf54l10_softdevice.hex`
  * :file:`s115_9.0.0-3.prototype_nrf54l05_softdevice.hex`

  For more details about this release of the SoftDevice, see :ref:`s115_docs`.

SoftDevice Handler
==================

* Added support for seeding the SoftDevice RNG when requested by the SoftDevice.

* Fixed the maximum MTU Kconfig option.

Boards
======

* Added the following ``bm_nrf54l15dk`` board variants for MCUboot (DFU-enabled board variants):

  * ``bm_nrf54l15dk/nrf54l05/s115_softdevice/mcuboot``
  * ``bm_nrf54l15dk/nrf54l10/s115_softdevice/mcuboot``
  * ``bm_nrf54l15dk/nrf54l15/s115_softdevice/mcuboot``

..

* Renamed the ``bm_nrf54l15dk`` board variants:

  * ``bm_nrf54l15dk/nrf54l05/softdevice_s115`` to ``bm_nrf54l15dk/nrf54l05/s115_softdevice``
  * ``bm_nrf54l15dk/nrf54l10/softdevice_s115`` to ``bm_nrf54l15dk/nrf54l10/s115_softdevice``
  * ``bm_nrf54l15dk/nrf54l15/softdevice_s115`` to ``bm_nrf54l15dk/nrf54l15/s115_softdevice``

..

* Fixed:

  * Fixed wrong RRAM size on the nRF54L10 emulation target from 1022 to 1012 KiB.

..

* Removed:

  * The template board variants.
  * The ``no_softdevice`` variants of the ``bm_nRF54l15dk``.

DFU
===

> * DFU based on MCUboot Loader mode is now introduced in the SDK:

Author: @nvlsianpu  pls check

      This ensures that log messages are printed and that the device goes into sleep mode.
      Returning from main leaves the device in a state that prevents low power sleep.
      Unhandled deferred log messages are not processed.

Bluetooth samples
-----------------

* Added the :ref:`ble_pwr_profiling_sample` sample.

..

* :ref:`ble_cgms_sample` sample:

  * Fixed an issue with responding twice to a PHY update request.

..

* :ref:`ble_hrs_sample` sample:

  * Removed the unused Coded PHY Kconfig fragment.

..

* :ref:`ble_nus_sample` sample:

  * Added support for Low Power UARTE (LPUARTE).

..

  * Fixed:

    * The pin configuration to use the pins from the :file:``board_config.h`` board header.
    * An issue in the :c:func:`ble_nus_data_send` function where passing NULL as the NUS instance could cause a segmentation fault.

Peripheral samples
------------------

* Added the following peripheral samples:

  * :ref:`bm_zms_sample` sample.
  * :ref:`bm_lpuarte_sample` sample.
  * :ref:`storage_sample` sample.

DFU samples
-----------

* Added the following samples:

>   * :ref:`ble_mcuboot_recovery_entry_sample` sample that demonstrates how to activate DFU mode on the device using a mobile application and execute DFU over Bluetooth LE.

Author: @nvlsianpu  pls check

.. _nrf_bm_release_notes_080:

Release Notes for |BMlong| v0.8.0
#################################

.. contents::
   :local:
   :depth: 2

|BMshort| v0.8.0 is the public launch release of this product, demonstrating a set of samples for the `nRF54L15 DK`_ with build targets supporting the following SoCs:

* `nRF54L15`_
* `nRF54L10`_ (emulated on the nRF54L15 DK)
* `nRF54L05`_ (emulated on the nRF54L15 DK)

The supported DK version is `nRF54L15 DK v0.9.3`_.

The release includes a prototype version of the S115 SoftDevice - a proprietary Bluetooth® LE stack to be used on the nRF54L15 DK.

Product maturity
****************

As this is version 0.8.0 of |BMshort|, please note that the solution is still under development and not production ready.

> While most modules and libraries are considered stable and are unlikely to undergo significant changes, adjustments might still be necessary.

Reviewer 3: I Agree, if any statement we should keep the expectations open to change. 
```suggestion
Modules and libraries are subject to changes.
```

This includes potential modifications to the organization of the folder structure.
Although the memory partition structure is expected to remain consistent, the sizes of individual partitions might be refined as progress is made towards a more stable release.

The documentation for |BMshort| is still in development and might not cover all libraries and modules available in the codebase.
Some sections might be incomplete or lacking detailed information as content is being developed.

IDE and tool support
********************

`nRF Connect extension for Visual Studio Code <nRF Connect for Visual Studio Code_>`_ is the recommended IDE for |BMshort| v0.8.0.
See the :ref:`install_nrf_bm` section for more information about supported operating systems and toolchain.

Supported boards
****************

* PCA10156 (`nRF54L15 DK`_)

Migration from nRF5 SDK
***********************

> For guidelines on how to migrate an application built on nRF5 SDK to |BMshort|, refer to `Migration notes - nRF5 SDK to nRF Connect SDK Bare Metal`_.

Reviewer 5: some links do not work (404 Page Not Found), I guess content will be added soon?

Author: yes, as soon as we tag

  For more details about this release of the SoftDevice, see :ref:`s115_docs`.

  For API documentation, see `S115 9.0.0-3.prototype API reference`_.

SoftDevice Handler
==================

* Added support for seeding the SoftDevice RNG when requested by the SoftDevice.

* Fixed the maximum MTU Kconfig option.

Boards
======

* Added the following ``bm_nrf54l15dk`` board variants for MCUboot (DFU-enabled board variants):

  * ``bm_nrf54l15dk/nrf54l05/s115_softdevice/mcuboot``
  * ``bm_nrf54l15dk/nrf54l10/s115_softdevice/mcuboot``
  * ``bm_nrf54l15dk/nrf54l15/s115_softdevice/mcuboot``

..

* Renamed the ``bm_nrf54l15dk`` board variants:

  * ``bm_nrf54l15dk/nrf54l05/softdevice_s115`` to ``bm_nrf54l15dk/nrf54l05/s115_softdevice``
  * ``bm_nrf54l15dk/nrf54l10/softdevice_s115`` to ``bm_nrf54l15dk/nrf54l10/s115_softdevice``
  * ``bm_nrf54l15dk/nrf54l15/softdevice_s115`` to ``bm_nrf54l15dk/nrf54l15/s115_softdevice``

..

* Fixed:

  * Fixed wrong RRAM size on the nRF54L10 emulation target from 1022 to 1012 KiB.

..

* Removed:

  * The template board variants.
  * The ``no_softdevice`` variants of the ``bm_nRF54l15dk``.

DFU
===

> * DFU based on MCUboot Loader mode is now introduced in the SDK:

Reviewer 7: ```suggestion
* DFU using MCUboot firmware loader mode is now introduced in the SDK:
```

>   * Experimental support added for DFU (update of Firmware Loader, SoftDevice, and application).

Reviewer 7: firmware loader, not capital

>     This is an asymmetric foreground DFU bank solution - the user application bank and SofDevice are much bigger than the update Loader.

Reviewer 7: *application and softdevice slots
*firmware loader slot

>   * This solution contains a port of the MCUmgr SMP protocol part required for performing the DFU operation.

Reviewer 7: a port? There is no port, it includes it as is, just say it uses MCUmgr to load firmware updates using image management.

Author: this point is now removed as per previous comments

>   * This solution contains a port of the MCUmgr SMP protocol part required for performing the DFU operation.

Reviewer 6: Name of this software module (both in software and documentation) is Firmware loader / firmware_loader

  For more details, see :ref:`bm_dfu`.

Logging
=======

* Fixed an issue with the minimal log mode.

Drivers
=======

* Added the following drivers:

  * The :ref:`driver_lpuarte` driver.
  * A SoftDevice-based flash driver implementing the Zephyr flash interface.

Libraries
=========

* Added the following libraries:

  * The :ref:`lib_bm_zms` library.
  * The ``bm_storage`` library.

>   * A queue library ported from Zephyr.

Reviewer 7: modified not ported, it's the exact same file with some lines removed, it was not ported to a new platform by recoding

Author: changed to
 * A queue library (modified library from Zephyr).

    This library has been added as a dependency for Firmware Loader and MCUmgr (used in DFU) offering support for threadless applications.

..

* ``ble_adv`` library:

  * Changed the default advertising name to ``nRF_BM_device``.
  * Corrected the :kconfig:option:`BLE_ADV_DIRECTED_ADVERTISING_HIGH_DUTY` Kconfig option.

..

* ``ble_conn_params library`` library:

  * Added:

    * Support for asymmetric data length.

>     * The updated connection parameter values to the ``BLE_CONN_PARAMS_EVT_UPDATED``` event.

Reviewer 7: is this an enum or what?
```
:c:enumerator:`BLE_CONN_PARAMS_EVT_UPDATED`
```

Reviewer 4: Yes. Enum.

  * Renamed slave latency to peripheral latency according to specification.

  * Fixed:

    * An issue where the ATT MTU could be set to a lower value than the current value.
    * An issue where the list of PHYs supported by the SoftDevice was not respected by the library when doing PHY updates.
    * Data length timeout issue and the :c:func:`data_length_get` function returning wrong value.
    * Incorrectly stated maximum ATT MTU value.

..

* ``bm_buttons`` library:

>   * Added the use of nrfx critical section API.

Author: it does seem like a trivial change so removing

  * Updated:

    * The main loop sleep instructions as the :c:func:`sd_app_evt_wait` function is deprecated.
    * Updated the main function flow to always end up in a loop with sleep instructions and deferred log processing instead of returning from the main function.
      This ensures that log messages are printed and that the device goes into sleep mode.
      Returning from main leaves the device in a state that prevents low power sleep.
      Unhandled deferred log messages are not processed.

Bluetooth samples
-----------------

* Added the :ref:`ble_pwr_profiling_sample` sample.

..

* :ref:`ble_cgms_sample` sample:

  * Fixed an issue with responding twice to a PHY update request.

..

* :ref:`ble_hrs_sample` sample:

  * Removed the unused Coded PHY Kconfig fragment.

..

* :ref:`ble_nus_sample` sample:

  * Added support for Low Power UARTE (LPUARTE).

..

  * Fixed:

    * The pin configuration to use the pins from the :file:``board_config.h`` board header.
    * An issue in the :c:func:`ble_nus_data_send` function where passing NULL as the NUS instance could cause a segmentation fault.

Peripheral samples
------------------

> * Added the following peripheral samples:

Reviewer 3: Either this or add DFU to the similar sentence below.
```suggestion
* Added the following samples:
```

  * :ref:`bm_zms_sample` sample.
  * :ref:`bm_lpuarte_sample` sample.
  * :ref:`storage_sample` sample.

DFU samples
-----------

* Added the following samples:

  * :ref:`ble_mcuboot_recovery_entry_sample` sample that demonstrates how to activate DFU mode on the device using a mobile application and execute DFU over Bluetooth LE.
  * :ref:`ble_mcuboot_recovery_retention_sample` sample that demonstrates how to switch a device into DFU mode on embedded application's request and execute DFU over UART or Bluetooth LE.

Known issues and limitations
============================

* Some issues are observed when using iPhone as the peer during testing of the Bluetooth samples.
* Some issues are observed when using Linux with the Bluetooth Low Energy app in nRF Connect for Desktop as the peer during testing of the Bluetooth samples.
* There are no samples or libraries available to demonstrate or support the Bluetooth LE Pairing and bonding functionality in the SoftDevice.
* The samples are not optimized for power consumption unless explicitly stated.

Documentation
=============

* First public release of product documentation.

..

> 

Reviewer 3: nit: extra whitespace

* Added:

  * Top-level sections for :ref:`drivers`, :ref:`libraries`, :ref:`bm_dfu`, and :ref:`s115_docs`.
  * Documentation for the :ref:`lib_ble_conn_params` library.
  * :ref:`nrf_bm_api` section.

> * :ref:`s115_docs` are now integrated into |BMshort| documentation rather than delivered as PDF files.

Reviewer 5: PDF files are still in bm repo, should we remove them going forward?

Author: We don't have to, some customers might prefer PDF. I will remove the 2nd part of the sentence.
*  :ref:`s115_docs` are now integrated into the |BMshort| documentation.
